server/migrations/helpers.js

```diff
@@ -1,50 +1,62 @@
 function calculatePreferences(user) {
-  const updatedPreferences = {
-    config: '',
-    sort: 0,
-    star: user?.preferences?.star || [],
-    complete: user?.preferences?.complete || [],
-  }
-  // Update complete
-  if (
-    !!user.preferences &&
-    user.preferences.complete &&
-    Object.keys(user.preferences.complete).length &&
-    !Array.isArray(user.preferences.complete)
-  ) {
-    updatedPreferences.complete = Object.keys(user.preferences.complete).map(
-      (siteName) => {
-        return {
-          site: siteName,
-          completedSubjects: user.preferences.complete[siteName],
-        }
+  const siteSubjects = []
+
+  if (!user.siteSubjects?.length) {
```

Comment:
I think there are ways to break up this function to make it more readable. I'm thinking something along the lines of

```js
calculatePreferences(user) {
  const { siteSubjects } = user
  const isUserMigrated = !!siteSubjects?.length
  const preferences = migratePreferences(user)

  if (userIsMigrated) {
    return { siteSubjects, preferences }
  } else {
    return { siteSubjects: migrateSiteSubjects(user), preferences }
  }
}
```